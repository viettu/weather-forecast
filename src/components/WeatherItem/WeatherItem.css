.WeatherItem-temperature {
    font-size: 15px;
    color: gray;
}