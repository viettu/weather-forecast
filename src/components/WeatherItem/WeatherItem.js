import React from 'react';
import PropTypes from 'prop-types';

import './WeatherItem.css';

import isoToDayWeek from '../../helpers/isoToDayWeek';
import formatTemperature from '../../helpers/formatTemperature';

const WeatherItem = ({applicableDate, minTemp, maxTemp}) => {
    return (
        <div className="WeatherItem-container">
            <h3>{isoToDayWeek(applicableDate)}</h3>
            <div className="WeatherItem-temperature">{`${formatTemperature(minTemp)} - ${formatTemperature(
                maxTemp
            )} â„ƒ`}</div>
        </div>
    );
};

WeatherItem.propTypes = {
    applicableDate: PropTypes.string,
    minTemp: PropTypes.number,
    maxTemp: PropTypes.number,
};

export default WeatherItem;
