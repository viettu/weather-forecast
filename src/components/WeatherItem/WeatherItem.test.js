import React from 'react';
import {render, screen, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom';

import WeatherItem from './WeatherItem';

afterEach(() => {
    cleanup();
});

const weatherItemData = {
    applicable_date: '2020-10-28',
    min_temp: 23.97,
    max_temp: 29.259999999999998,
};

test('renders without crashing', () => {
    render(<WeatherItem />);
});

test('snapshot testing', () => {
    let renderResult = render(<WeatherItem />);
    expect(renderResult.container.firstChild).toMatchSnapshot();

    renderResult = render(
        <WeatherItem
            applicableDate={weatherItemData.applicable_date}
            minTemp={weatherItemData.min_temp}
            maxTemp={weatherItemData.max_temp}
        />
    );
    expect(renderResult.container.firstChild).toMatchSnapshot();
});

test('display information', async () => {
    render(
        <WeatherItem
            applicableDate={weatherItemData.applicable_date}
            minTemp={weatherItemData.min_temp}
            maxTemp={weatherItemData.max_temp}
        />
    );

    expect(await screen.findByText('Wednesday')).toBeVisible();
    expect(await screen.findByText('24.0 - 29.3 â„ƒ')).toBeVisible();
});
