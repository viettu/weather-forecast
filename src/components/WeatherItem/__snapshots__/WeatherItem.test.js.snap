// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<div
  class="WeatherItem-container"
>
  <h3 />
  <div
    class="WeatherItem-temperature"
  >
     -  ℃
  </div>
</div>
`;

exports[`snapshot testing 2`] = `
<div
  class="WeatherItem-container"
>
  <h3>
    Wednesday
  </h3>
  <div
    class="WeatherItem-temperature"
  >
    24.0 - 29.3 ℃
  </div>
</div>
`;
